content/authoring/kata/floating-point.md

```diff
@@ -0,0 +1,220 @@
+
+The article is not meant to be a comprehensive guide explainign what are floating-point numbers in general, how they are represented, how they work internally, etc. It's meant to be targeted mostly to authors of Codewars kata to help them create content of better quality and avoid traps and pitfalls leading to problems with precision loss, comparisons, etc.
+
+
+## What are floating-point numbers?
+
+Floating-point numbers come in many different shapes and colors, but one (or, if you prefer, two) of them is the most widespread, the most common, and, at the same time, the most problematic for newbies (and not only). Whenever you see or use anything called _"float"_, _"double"_, _"single-"_ or _"double-precision floating point number"_, _"IEEE-754 number"_, you need to be careful, because things can often turn out surprising and not as easy as one might think. Through the rest of the article, the term _"floating-point numbers"_ will refer specifically to **IEEE-754 double-precision values**, unless explicitly stated otherwise.
+
+Floating-point numbers are usually not specific to any particular language, system, or architecture. IEEE-754 numbers are widely adopted and used across many platforms, languages, and compilers. With some small exceptions, anything written in this article applies in the same way to Python, C, JavaScript, or Haskell.
+
+<details>
+
+<summary>"Why"</summary>
+
+## Strange things about floating-point values
+
+Floating-point numbers are useful, because they have capabilities not available to other simple types available in majority of languages:
+
+- They can hold fractional values, like `25.5` or `-7.125`. Fractions can be as small as ~1.7\*10<sup>-308</sup>
+- They have much wider range than integral types of the same size. Signed 64-bit integer data type can hold values up to ~9\*10<sup>18</sup>, while 64-bit floating-point value can go up to ~1.7\*10<sup>308</sup>.
+
+However, to achieve these goals, floating-point values trade off their _precision_, what more or less means that there are some values which cannot be represent exactly. They are capable of storing up to 52 bits, or approximately 15 decimal digits, and anything beyond this is lost. Values which would require better precision are internally rounded and stored just as approximations. And this leads to many problems which often can come up as surprising.
+
+### Representability
+
+As it was already mentioned, precision of a floating-point value is limited to 52 bits, or approx. 15 decimal digits. This means that some values simply cannot be stored in a floating-point variable. For example, value `10000000000000000` (or 1*10<sup>16</sup>) can be represented exactly, but there's no `10000000000000001`! How do you think, what will following code print?
```

Comment:
I will think of some better example if we decide this part to stay.

Do you think it's worth to keep the _"Strange things about floating-point values"_ part, or just remove it and keep only _"How to avoid problems with floating-point numbers?"_

Reply:
IDK, although these things aren't really strange, they are more like possibly unexpected.

Reply:
The "why" part was redone, and greatly simplified. There's no more "strange things" there, and troublesome example above was removed.

I am still not sure whether to keep it, but now it's at least reduced.